en/developer_guide/best_practice_js.rst

```diff
@@ -0,0 +1,288 @@
+
+
+
+
+Tips :
+------
+Don't Use new Object()
+Use {} instead of new Object()
+Use "" instead of new String()
+Use 0 instead of new Number()
+Use false instead of new Boolean()
+Use [] instead of new Array()
+Use /()/ instead of new RegExp()
+Use function (){} instead of new function()
+
+.. code-block:: js
+
+    var x1 = {};           // new object
+    var x2 = "";           // new primitive string
+    var x3 = 0;            // new primitive number
+    var x4 = false;        // new primitive boolean
+    var x5 = [];           // new array object
+    var x6 = /()/;         // new regexp object
+    var x7 = function(){}; // new function object
+
+Function declarations within blocks:
+------------------------------------
+
+Do not do this:
+
+.. code-block:: js
+
+    if (x) {
+      function foo() {}
+    }
+
+While most script engines support Function Declarations within blocks it is not part of
+ECMAScript (see ECMA-262, clause 13 and 14). Worse implementations are inconsistent with
+each other and with future EcmaScript proposals. ECMAScript only allows for Function
+Declarations in the root statement list of a script or function. Instead use a variable
+initialized with a Function Expression to define a function within a block:
+
+Namespace:
+----------
+
+The first step to good JavaScript object management is having a namespace, or a JavaScript
+object that contains our code and data, that you know will not conflict with other extensions.
+
+.. code-block:: js
+
+    /**
+     * myNamespace namespace.
+     */
+    if (typeof myNamespace == "undefined") {
+        var myNamespace = {};
+
+        foo: function() {
+        },
+
+        bar: function() {
+        }
+    };
+
+    myNamespace.foo();
+
+
+Exception and custom exception:
+-------------------------------
+
+You basically can't avoid exceptions if you're doing something non-trivial. go for it.
+
+Without custom exceptions, returning error information from a function that also returns
+a value can be tricky, not to mention inelegant. Bad solutions include passing in a
+reference type to hold error information or always returning Objects with a potential error
+member. These basically amount to a primitive exception handling hack. Feel free to use
+custom exceptions when appropriate.
+
+Multi level prototype:
+----------------------
+
+Not preferred by Google. These hierarchies are much harder to get right than they first appear!
+For that reason, it is best to use `goog.inherits()` from the Closure Library or a similar
+library function.
+
+Source and other documents:
+---------------------------
+
+- Conventions:
+ - Document proposed by google : https://google.github.io/styleguide/javascriptguide.xml
+ - Other document produced by the comunity : https://github.com/rwaldron/idiomatic.js
+ (*different tools to validate as jshint*)
+
+- Comment js (example http://usejsdoc.org/)
+- Namespace : on Open Orchestra a big part of js and the global scope, a integrator may
```

Comment:
I think a verb is missing in the first part of the sentence.
+ an integrator rather than a integrator